grails-app/controllers/org/pih/warehouse/core/PaymentTermController.groovy

```diff
@@ -11,5 +11,58 @@ package org.pih.warehouse.core
 
 class PaymentTermController {
 
-    def scaffold = PaymentTerm
+    def list = {
+        params.max = Math.min(params.max ? params.int('max') : 10, 100)
+        [paymentTerms: PaymentTerm.list(params), paymentTermsTotal: PaymentTerm.count()]
+    }
+
+    def create = {
+        def paymentTerm = new PaymentTerm()
+        paymentTerm.properties = params
+        return [paymentTerm: paymentTerm]
+    }
+
+    def edit = {
+        def paymentTerm = PaymentTerm.get(params.id)
+        return [paymentTerm: paymentTerm]
+    }
+
+    def save = {
+        def paymentTerm = PaymentTerm.get(params.id)
+        if (paymentTerm) {
+            paymentTerm.properties = params
+            if (!paymentTerm.hasErrors() && paymentTerm.save(flush: true)) {
```

Comment:
hasErrors check is not required

Reply:
The hasErrors check is fine (Grails generates the controllers this way). The point I was making with your PR was that the validate() shouldn't have been needed because it happens during the binding above.